pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/SegmentPreprocessingJob.java

```diff
@@ -446,6 +480,21 @@ private Schema getSchema(Path inputPathDir)
     return avroSchema;
   }
 
+  private long countAvroRecords(List<Path> inputPaths) throws IOException {
+    long count = 0;
+
+    for (Path path : inputPaths) {
+      DataFileStream<GenericRecord> dataFileStream = getAvroReader(path);
```

Comment:
We need to scan all the files to get the record count? That will be super expensive

Reply:
this seems to be the best approach from my research, do you have other suggestions?